### PR TITLE
Implemented Velocikey config

### PR DESCRIPTION
**Pull request by tifrel**
[As stated in the docs](https://docs.qmk.fm/#/feature_velocikey), Velocikey is currently not configurable without hacking inside the `quantum` dir. This PR aims to fix that via simple macros that can be defined in a custom `config.h`.

## Description

Just added configuration options for Velocikey, such as acceleration, decay rate, or the min/max timings per animation. Updated docs as well. However if this doesn't pass Travis, I would like some help because I had some compilation issues on boards/layouts that I don't have, so I am not really certain how to troubleshoot this. However maybe the fails are not reproducible.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by tillyboy** [comment]
Now that it doesn't pass CI, I would like at least some feedback. I reviewed the travis report and I cannot spot the connection between my changes and the fails.

File: quantum/rgblight.h [hunk 1]
```diff
@@ -170,6 +170,45 @@ enum RGBLIGHT_EFFECT_MODE {
 #        define RGBLIGHT_LIMIT_VAL 255
 #    endif
 
+// velocikey customization
```
Comment (drashna) [inline quantum/rgblight.h:173]: If anything, I'd say make these non-specific to velocikey, and make them just global rgblight settings.  (eg, change the names to reflect rgblight, rather than velocikey).

But that may be better as a separate PR.

Comment (tillyboy) [reply quantum/rgblight.h:173]: My reasoning was that, at least from my understanding of the code, these variables don't have any effect when not used with Velocikey. But maybe I got it wrong, and could of course rename it.

File: docs/feature_velocikey.md [hunk 2]
```diff
@@ -16,15 +17,43 @@ VELOCIKEY_ENABLE = yes
...
 
-Support for LED breathing effects is planned but not available yet.
+-   RGB Breathing
+-   RGB Rainbow Mood
+-   RGB Rainbow Swirl
+-   RGB Snake
+-   RGB Knight
 
- As long as Velocikey is enabled, it will control the speed regardless of any other speed setting that your RGB lights are currently on.
+Support for LED breathing effects is planned but not available yet.
```
Comment (spidey3) [inline docs/feature_velocikey.md:27]: Do you mean by this LED Backlight?

Comment (tillyboy) [reply docs/feature_velocikey.md:27]: I think this is a glitch by `diff`, because all I did was remove line 30, as I added the configurability via `config.h`, as documented below.

File: quantum/rgblight.h (hunk 1 above)
Comment (tillyboy) [reply quantum/rgblight.h:173]: Any updates on this?

File: docs/feature_velocikey.md [hunk 3]
```diff
@@ -16,15 +17,43 @@ VELOCIKEY_ENABLE = yes
...
- - RGB Rainbow Swirl
- - RGB Snake
- - RGB Knight
 
-Support for LED breathing effects is planned but not available yet.
+-   RGB Breathing
+-   RGB Rainbow Mood
+-   RGB Rainbow Swirl
+-   RGB Snake
+-   RGB Knight
```
Comment (fauxpark) [inline docs/feature_velocikey.md:25]:
```suggestion
 - RGB Breathing
 - RGB Rainbow Mood
 - RGB Rainbow Swirl
 - RGB Snake
 - RGB Knight
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
This has some merge conflicts.

**Comment by tifrel** [comment]
Ping for review then, I still think this is a valid enhancement, and saves others the hassle of having to figure it out by themselves.

**Comment by fauxpark** [comment]
You should be able to reopen the PR yourself - you will also need to switch the base branch (which might require a rebase) as  all core changes now go through `develop`.
